testsuite/httpx/__init__.py

```diff
@@ -18,15 +20,50 @@ def create_tmp_file(content: str):
     return file
 
 
-class UnexpectedResponse(Exception):
-    """Slightly different response attributes were expected or no response was given"""
+class KuadrantResult:
```

Comment:
Just an idea for discussion. Change this to `KuadrantResponse` and extend `Response`.

Reply:
I can change the name, but sadly there is no way how I can extend the class, because in case of succes response I would have to do basically a copy (pass all params from response I got into a new object) and in case of error I wouldnt have all the required info to create a response